tests/unit/processor/pseudonymizer/test_pseudonymizer.py

```diff
@@ -255,534 +857,143 @@ def test_pseudonymize_two_fields(self):
+        assert len(extra_output) == 2
+        assert isinstance(extra_output[0], list)
+        assert isinstance(extra_output[1], tuple)
+        assert isinstance(extra_output[1][0], dict)
+        assert extra_output[1][0] == {"kafka": "topic"}, "Output is setted as in CONFIG"
+        assert extra_output[0][0].get("pseudonym"), "pseudonym is set"
+        assert extra_output[0][0].get("origin"), "encrypted origional is set"
+        assert extra_output[0][0].get("@timestamp"), "timestamp is set if present in event"
+
+    def test_ignores_missing_field(self):
+        rule_dict = {
+            "filter": "winlog.event_id: 1234 AND winlog.provider_name: Test456",
             "pseudonymizer": {
                 "pseudonyms": {
-                    "pseudo_this": regex_pattern,
+                    "does_not_exists": "RE_WHOLE_FIELD",
+                    "winlog.event_data.param2": "RE_WHOLE_FIELD",
                 }
             },
         }
-        self._load_specific_rule(rule)
-        self.object.process(event)
-
-        assert event["pseudo_this"] == pseudonym
-
-    def test_pseudonymize_one_element_from_list_with_two_different_elements(self):
-        pseudonym = [
-            "foo\\<pseudonym:d95ac3629be3245d3f5e836c059516ad04081d513d2888f546b783d178b02e5a>",
-            "bar",
-        ]
-
-        regex_pattern = "RE_DOMAIN_BACKSLASH_USERNAME"
         event = {
-            "pseudo_this": ["foo\\test", "bar"],
-        }
-        rule = {
-            "filter": "pseudo_this",
-            "pseudonymizer": {
-                "pseudonyms": {
-                    "pseudo_this": regex_pattern,
-                }
+            "@timestamp": "custom timestamp",
+            "winlog": {
+                "event_id": 1234,
+                "provider_name": "Test456",
+                "event_data": {
+                    "param1": "Pseudonymize me!",
+                    "param2": "Pseudonymize me!",
+                },
             },
         }
-        self._load_specific_rule(rule)
-        self.object.process(event)
-
-        assert event["pseudo_this"] == pseudonym
-
-    def test_replace_regex_keywords_by_regex_expression_can_be_called_multiple_times(self):
-        rule_dict = {
-            "filter": "event_id: 1234",
-            "pseudonymizer": {"pseudonyms": {"something": "RE_WHOLE_FIELD"}},
-            "description": "description content irrelevant for these tests",
-        }
         self._load_specific_rule(rule_dict)  # First call
-        expected_pattern = re.compile("(.*)")
-        assert self.object._specific_tree.rules[0].pseudonyms == {"something": expected_pattern}
-        self.object._replace_regex_keywords_by_regex_expression()  # Second Call
-        assert self.object._specific_tree.rules[0].pseudonyms == {"something": expected_pattern}
+        extra_output = self.object.process(event)
+        assert extra_output[0][0].get("pseudonym"), "pseudonym is set"
```

Comment:
Maybe you can add here an assertion that checks that there is also only one pseudonym? You check `extra_output[0][0]` but there could be still also an `extra_output[1][0]` :wink:

Reply:
all pseudonyms are in the zero index... in the index 1 are the desired outputs.

Reply:
but I added the check